bundles/org.eclipse.search/search/org/eclipse/search/internal/ui/SearchDialog.java

```diff
@@ -174,6 +174,8 @@ public SearchDialog(IWorkbenchWindow window, String pageId) {
 
 		fPageChangeListeners= null;
 		setUseEmbeddedProgressMonitorPart(false);
+		setShellStyle(getShellStyle() ^ SWT.APPLICATION_MODAL | SWT.MODELESS);
+		setBlockOnOpen(false);
```

Comment:
Honestly I don't know what effect this call will have. I tested with true and false but didn't see any visible difference (I largely derived this logic from FindReplaceDialog). I thought I will ask here.

Reply:
The "blockOnOpen" property sets whether the method that opens the dialog expects the dialog to block execution until the dialog was closed or whether the opening method does not expect that

http://www.java2s.com/example/java-api/org/eclipse/jface/dialogs/dialog/setblockonopen-1-0.html